FinancialTransactionsApi.Tests/V1/Boundary/Response/ApiResponseGenericTypeTests.cs

```diff
@@ -0,0 +1,51 @@
+using AutoFixture;
+using FinancialTransactionsApi.V1.Boundary.Request;
+using FinancialTransactionsApi.V1.Boundary.Response;
+using FinancialTransactionsApi.V1.Domain;
+using FinancialTransactionsApi.V1.Helpers;
+using FluentAssertions;
+using Hackney.Core.DynamoDb;
+using System.Collections.Generic;
+using Xunit;
+
+namespace FinancialTransactionsApi.Tests.V1.Boundary.Response
+{
+    public class ApiResponseGenericTypeTests
+    {
+        private readonly Fixture _fixture = new Fixture();
+
+        [Fact]
+        public void ApiResponseGenericType_ReturnTheSameObjectType()
+        {
+            var transactionsList = _fixture.Build<Transaction>().Create();
+
+            var responseWrapperMockObject = new ResponseWrapper<Transaction>(transactionsList);
+
+            ApiResponse<ResponseWrapper<Transaction>> request = new ApiResponse<ResponseWrapper<Transaction>>(responseWrapperMockObject);
+
+            request.Results.Should().BeEquivalentTo(responseWrapperMockObject);
+        }
+
+        [Fact]
+        public void ApiResponseGenericType_ReturnTheSameCollectionOfObjectTypes()
+        {
+            var transactionsList = _fixture.Build<Transaction>().CreateMany();
+
+            var responseWrapperMockObject = new ResponseWrapper<IEnumerable<Transaction>>(transactionsList);
+
+            ApiResponse<ResponseWrapper<IEnumerable<Transaction>>> request = new ApiResponse<ResponseWrapper<IEnumerable<Transaction>>>(responseWrapperMockObject);
+
+            request.Results.Should().BeEquivalentTo(responseWrapperMockObject);
+        }
+
+        [Fact]
+        public void ApiResponseGenericType_ReturnNull()
+        {
+            ApiResponse<ResponseWrapper<IEnumerable<Transaction>>> request = new ApiResponse<ResponseWrapper<IEnumerable<Transaction>>>();
+
+            request.Total.Should().Be(default(long));
+
+            request.Results.Should().BeNull();
```

Comment:
This is testing for the Results to be null.
If there are no results, should be an empty array => `results = [ ]`

So the test should be positive for 
`request.Results.Should().BeEmpty();`

Reply:
in this test case i use a default constructor which doesn't initialize the Result property, in this situation the default value will be null, by the way this comment reminds me to create another test case for an empty collection